e2e/op_stack.go

```diff
@@ -0,0 +1,282 @@
+package e2e
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-batcher/batcher"
+	"github.com/ethereum-optimism/optimism/op-batcher/compressor"
+	opbatchermetrics "github.com/ethereum-optimism/optimism/op-batcher/metrics"
+	opnodemetrics "github.com/ethereum-optimism/optimism/op-node/metrics"
+	opnode "github.com/ethereum-optimism/optimism/op-node/node"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/driver"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/sync"
+	opproposermetrics "github.com/ethereum-optimism/optimism/op-proposer/metrics"
+	"github.com/ethereum-optimism/optimism/op-proposer/proposer"
+	opcrypto "github.com/ethereum-optimism/optimism/op-service/crypto"
+	"github.com/ethereum-optimism/optimism/op-service/dial"
+	"github.com/ethereum-optimism/optimism/op-service/sources"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/polymerdao/monomer/utils"
+	"github.com/sourcegraph/conc"
+)
+
+type OPEventListener interface {
+	// LogWithPrefix is meant to be used as a general-purpose log function.
+	// The prefix will never be empty and r will never be nil.
+	// It is only used to fulfill log interfaces defined by third-party OP Stack components.
+	LogWithPrefix(prefix string, r *log.Record)
+	AfterStartup()
+	BeforeShutdown()
+}
+
+type OPStack struct {
+	// The http address (without the http://) of the Ethereum EL node.
+	l1Address string
+	// The websocket address (without the ws://) of the execution engine.
+	engineAddress string
```

Comment:
maybe, for consistency w/ other listings of these addresses:
```suggestion
    // the websocket address (without the we://) of the Monomer execution engine.
	monomerAddress string
```

Reply:
Looking at this again, I think "execution engine" is sufficient. The OP CL is agnostic to the engine implementation.